### PR TITLE
added curaleaf spider

### LINKED ISSUE #1173: Curaleaf
https://curaleaf.com/locations/

### PR DESCRIPTION
**Pull request by asharma13524**
Closes #1173 

Added spider for curaleaf mention here https://github.com/alltheplaces/alltheplaces/issues/1173

Still a bit of a beginner so please let me know if I did something incorrectly. 

Also wondering what the main motivations/goals for the repository are? What can all this data ultimately be used for? 

Thanks.

### CODE REVIEW

**Comment by asharma13524** [comment]
Sorry, still working on this.

File: locations/spiders/curaleaf.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                'street': street,
+                'city': city,
+                'state': state,
+                'postcode': postcode,
+                'country': country,
+                'addr_full': address
+            }
+            yield GeojsonPointItem(**properties)
+
+        except error:
```
Comment (iandees) [inline locations/spiders/curaleaf.py:58]: Please don't wrap the entire function in a try/except. We should see those errors and fix them if they come up.

File: locations/spiders/curaleaf.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+
+class CuraleafSpider(scrapy.Spider):
+    name = "curaleaf"
+    allowed_domains = ["curaleaf.com"]
+    start_urls = (
+        'https://curaleaf.com/locations/',
+    )
+
+    def phone_normalize(self, phone):
```
Comment (iandees) [inline locations/spiders/curaleaf.py:15]: It doesn't seem like you're calling this anywhere. Can you remove it if you aren't using it?

Comment (asharma13524) [reply locations/spiders/curaleaf.py]: my latest commit should have addressed these. let me know

**Pull request merged**